dxr/app.py

```diff
@@ -162,6 +164,27 @@ def _tree_tuples(trees, tree, query_text, is_case_sensitive):
              description)
             for t, description in trees.iteritems()]
 
+@dxr_blueprint.route('/<tree>/raw/<path:path>')
+def send_raw(tree, path):
+    """Send raw data at path from tree, for binary things like images."""
+    query = {
+            'filter': {
+                'term': {'path': path}
+                }
+            }
+    index = current_app.config['ES_ALIASES'][tree]
+    results = current_app.es.search(
+            query,
+            index=index,
+            doc_type=FILE, size=1)
+    try:
+        # we explicitly get index 0 because there should be exactly 1 result
+        data = results['hits']['hits'][0]['_source']['raw_data'][0]
+        dataFile = StringIO(data.decode("base64"))
```

Comment:
Nits! :-) `dataFile` should be `data_file` for conformance with PEP 8's naming conventions. And let's go with single quotes, as we typically do when we're not trying to enclose other single quotes.

More substantially, we should keep the body of `try` statements minimal so we don't catch errors we weren't expecting. So let's move the second two lines of this one to after the `except`. You can put it in an `else` clause if you like, or just leave that out, since your `except` exits the function anyway.